lib/x/response_parser.rb

```diff
@@ -39,9 +39,12 @@ class ResponseParser
     def parse(response:, array_class: nil, object_class: nil)
       raise error(response) unless response.is_a?(Net::HTTPSuccess)
 
-      return unless json?(response)
+      return if response.instance_of?(Net::HTTPNoContent)
 
-      JSON.parse(response.body, array_class:, object_class:)
+      begin
+        JSON.parse(response.body, array_class:, object_class:)
+      rescue JSON::ParserError # rubocop:disable Lint/SuppressedException
```

Comment:
If the X API returns invalid JSON, this library should raise an error. If users want to ignore that error, they can do so at the application level.

Reply:
This is related to the comment I did here : https://github.com/sferik/x-ruby/pull/38#discussion_r1927395463

Previously, we relied on headers to parse JSON but now, there is a request made with "application/octet-stream" which contains a valid JSON. As before, `nil` should be returned when the body does not contain a JSON.